src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationUtilities.cs

```diff
@@ -109,8 +111,10 @@ internal static class SemanticClassificationUtilities
                 {
                     var classifiedSpans = ClassificationUtilities.GetOrCreateClassifiedSpanList();
 
-                    await classificationService.AddSemanticClassificationsAsync(
-                        document, snapshotSpan.Span.ToTextSpan(), classifiedSpans, cancellationToken: cancellationToken).ConfigureAwait(false);
+                    var newSpans = await classificationService.GetSemanticClassificationsAsync(
+                        document, snapshotSpan.Span.ToTextSpan(), cancellationToken).ConfigureAwait(false);
+
+                    CombineSpans(classifiedSpans, newSpans);
```

Comment:
Agreed. All [ClassificationUtilities.GetOrCreateClassifiedSpanList()](http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/Classification/ClassificationUtilities.cs,e76d856b7ebe99d4) does is return a cached, empty `List<ClassifiedSpan>` or create a new one. There shouldn't be anything to combine. Really, we shouldn't need this at all since we're choosing to allocate here rather than pass a list in.